### PR TITLE
chore: Add script to surface-scan components against specification

### LINKED ISSUE #8969: Build script to scan sources for component specification issues
A number of the conditions listed in the [component specification](https://github.com/timberio/vector/blob/master/docs/specs/component.md) #8728 may be statically checked, at least at a surface level, by a script similar to https://github.com/timberio/vector/blob/master/scripts/check-events.sh We should replace `check-events.sh` with such a script to enforce better uniformity in components.

### PR DESCRIPTION
**Pull request by bruceg**
This is resurrected from #9110 limited to just checking events (as well as duplicating the functionality of `scripts/check-events.sh`). This again reveals too many issues to be useful (67 at current count), but would still be useful at directing efforts and preventing regressions in the future.

Closes #8969 

Signed-off-by: Bruce Guenter <bruce.guenter@datadoghq.com>

### CODE REVIEW

**Comment by bruceg** [comment]
Amusingly, this identified two issues that the existing `scripts/check-events.sh` should have errored on, but didn't:
```
src/config/mod.rs: Errors in message "`healthcheck_uri` option has been deprecated, use `healthcheck.uri` instead. ":
  Message must start with a capital.
  Message must end with a period.
src/sources/host_metrics/filesystem.rs: Errors in message "Failed to load partitions info":
  Message must end with a period.
```

**Review by blt — APPROVED** [review]
I can't comment specifically on the implementation because I don't understand ruby. But, if it's got a better hit rate than our existing checks I'm for it.

File: src/internal_events/apache_metrics.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// ## skip check-events ##
```
Comment (jszwedko) [inline src/internal_events/apache_metrics.rs:1]: Nice 👍

File: scripts/check-events [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+require 'find'
+
+COMMON_EVENTS = ['EventsReceived', 'EventsSent']
+# Sources implicitly emit EventsSent in the common pipeline code
+SOURCE_EVENTS = ['BytesReceived'] + COMMON_EVENTS - ['EventsSent']
+# Sinks implicitly emit EventsReceived in the common pipeline code
+SINK_EVENTS = COMMON_EVENTS + ['BytesSent'] - ['EventsReceived']
+# Transforms implicitly emit the common events in the topology, and have
+# no other emit requirements.
+TRANSFORM_EVENTS = []
```
Comment (jszwedko) [inline scripts/check-events:13]: Are these used anywhere? I'm not seeing it.

File: scripts/check-events [hunk 3]
```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/ruby
+# coding: utf-8
+
+require 'find'
```
Comment (jszwedko) [inline scripts/check-events:4]: Does this need to be added to `scripts/Gemfile`?

**Review by jszwedko — APPROVED** [review]
Nice! I just left a few comments, but this looks good to me.

File: scripts/check-events [hunk 4]
```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/ruby
```
Comment (jszwedko) [inline scripts/check-events:1]:
```suggestion
#!/usr/bin/env ruby
```

I think this is more common.

File: scripts/check-events (hunk 2 above)
Comment (bruceg) [reply scripts/check-events:13]: Oh, oops, those were part of the old script that I didn't purge out. Fixed in f2e1a49c4

File: scripts/check-events (hunk 4 above)
Comment (bruceg) [reply scripts/check-events:1]: Fixed in a6e5db7d8

File: scripts/check-events (hunk 3 above)
Comment (bruceg) [reply scripts/check-events:4]: I have no idea. It's part of the base ruby install on my system, so I _think_ not.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**